### PR TITLE
feat: sendBeacon

### LINKED ISSUE #14307: Interest in Adding Support for "navigator.sendBeacon"?
I was following up on trying to use OpenTelemetry-JS in Deno (after my earlier efforts [here](https://github.com/open-telemetry/opentelemetry-js/issues/2293#issuecomment-1030750431)) this time trying to export trace data out of Deno and to an OTEL collector.

The main issue I ran into this time was that the [browser version of the exporter](https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/exporter-trace-otlp-http/src/platform/browser/OTLPExporterBrowserBase.ts#L80) relies on either navigator.sendBeacon or XmlHttpRequest to do the exporting, neither of which are currently available in Deno.

I was able to put together this [small userland polyfill](https://deno.land/x/sendbeacon_polyfill@0.0.1) of navigator.sendBeacon based on [this comment from MDN](https://developer.mozilla.org/en-US/docs/Web/API/fetch#keepalive) about the "keepalive" option for fetch, which seemed to do the trick as far as my OTEL-JS experiment goes. However I'm not sure if its description there as a "drop-in replacement for navigator.sendBeacon" also applies in Deno.

I remember seeing elsewhere XmlHttpRequest won't be supported in core, but I was curious if the same would apply to navigator.sendBeacon? I'm not quite sure what the direction is around navigator APIs, as I see there seem to only be 2 supported at the moment (?) so I figured it couldn't hurt to ask.

### PR DESCRIPTION
**Pull request by load1n9**
sendBeacon api implementation 
spec: https://w3c.github.io/beacon/#sendbeacon-method
Closes #14307

### CODE REVIEW

File: ext/fetch/26_fetch.js [hunk 1]
```diff
@@ -580,7 +581,37 @@
     }
   }
 
+  /**
+   * @param {string} url
+   * @param {BodyInit} data
+   */
+  function sendBeacon(url, data) {
+    const prefix = "Failed to construct 'Beacon'";
```
Comment (crowlKats) [inline ext/fetch/26_fetch.js:589]:
```suggestion
    const prefix = "Failed to execute 'sendBeacon'";
```

**Comment by crowlKats** [comment]
sadly due to the fact that the wpts use iframes, tests will have to be written manually

**Comment by aapoalas** [comment]
One WPT test failing, also needs rebase now.

**Comment by crowlKats** [comment]
yea, i have to look into the wpt, will try to make it for the release

**Comment by lino-levan** [comment]
@load1n9 could you rebase?

**Review by aapoalas — CHANGES_REQUESTED** [review]
**issue:** I realized that the spec requires `sendBeacon` to return a synchronous boolean value, and not a Promise that resolves with a boolean like here.

Properly implementing this API will require binding into the Rust side of the HTTP implementation.

**Comment by lino-levan** [comment]
My understanding of the spec is that it DOES returns a synchronous boolean value, but it sends the actual data asynchronously. It returns `true` after the data has been queued (after fetch is constructed in this case?). I don't think we need rust bindings for this functionality.

**Review by lino-levan — CHANGES_REQUESTED** [review]
Two big things:
- `sendBeacon` returns a promise when it should not be doing so
- `sendBeacon` should set `keepAlive` which it doesn't seem to be doing

**Pull request closed without merging**